src/v/cluster/archival_metadata_stm.h

```diff
@@ -16,7 +16,7 @@
 #include "cluster/persisted_stm.h"
 #include "model/metadata.h"
 #include "model/record.h"
-#include "utils/mutex.h"
+#include "ssx/semaphore.h"
```

Comment:
> util: move mutex into ssx/semaphore.h

i'm not opposed to the move, but i'm also not sure why it should move. similarly, i think stuff in ssx could easily move to utils. originally i think was intended to be seastar extensions, but this just ends up looking like utilities at some point.

Reply:
Following @BenPope suggestion here. I'm cool with either, but I think having them together is good.

Reply:
> Following @BenPope suggestion here. I'm cool with either, but I think having them together is good.

I guess an ntp often falls outside of 15 bytes. It's probably not a big deal in this case, I was just discussing interface design in general. Death by a thousand cuts and all that.